lib/model.js

```diff
@@ -2559,6 +2559,9 @@ class Model {
       if (options.updateOnDuplicate && (dialect !== 'mysql' && dialect !== 'mariadb' && dialect !== 'sqlite' && dialect !== 'postgres')) {
         throw new Error(`${dialect} does not support the updateOnDuplicate option.`);
       }
+      if (options.updateOnDuplicate && options.where && dialect !== 'postgres') {
+        throw new Error(`${dialect} does not support the where clause option for bulkCreate.`);
+      }
```

Comment:
Both sqlite and postgres support this.

Also;

There's 2 clauses which might show up for both dialects in an `ON CONFLICT DO UPDATE` statement: 
- `ON CONFLICT WHERE` - This is used for indices, only references to the target table are allowed. E.g. `ON CONFLICT WHERE "username"` vs `ON CONFLICT WHERE "users"."username"`.
- `ON CONFLICT DO UPDATE WHERE` - This doesn't help with indices - however it allows for more advanced clause for when to update.

Reply:
Oh nvm, its actually `onConflictDoUpdateWhere` not `where`.  Might be worth simplifying this term, I'm using `conflictWhere` for the first `WHERE`.